### ISSUE TITLE
Preview cookie persisting unexpectedly

### ISSUE THREAD

**Issue by NeversSync** [prismicio/prismic-gatsby#477]
Thank you for all the hard work on the recent upgrades, much appreciated!

### Versions

- gatsby-source-prismic: 5.1.0
- gatsby-plugin-prismic-previews: 5.0.3
- node: 14.17.0

### Reproduction

Package.json: 
![Screen Shot 2021-11-05 at 11 38 50 AM](https://user-images.githubusercontent.com/20482636/140561985-1d05d954-7026-4892-9ffc-b25b9d561805.png)

## My general configuration:
- I have the `PrismicPreviewProvider` with `repositoryConfig` set only in `wrapRootElement` within gatsby-browser.js and gatsby-ssr.js, and each document template file is using `withPrismicPreview({document type})` with no `repositoryConfig`. 
- Each of the template file queries is using the `_previewable` field
- 404.js page is using `withPrismicUnpublishedPreview(NotFound)`
- preview.js page is using `withPrismicPreviewResolver(PreviewPage)`

### Steps to reproduce
- Follow the instructions for V5 of source plugin and preview config. 
- Open a preview from within the Prismic UI on a published, or unpublished document. I am opening the preview on a staging site (beta.site.com). I have also tested this on local dev server. Previews work ok so far. Toolbar loads, preview cookie is visible in application tab (chrome browser)
- Close the tab with the preview
- Open a new tab or window with same domain name (beta.site.com, I am also seeing this issue if I open production at site.com)

### What is expected?
- 'Fetch Preview' popup and preview toolbar should only load when preview is triggered through the Prismic UI. 
- Opening new browser tabs or windows with similar domain names should not trigger the fetch preview pop up or toolbar to load. 
- The cookie that I presume is being set and is visibly set to expire per `session` should be isolated to single browser tab or session.

### What is actually happening?
- Preview cookie is persisting and not being removed. Toolbar and fetch preview popup open and persist until I either click on the X on the toolbar or manually delete the cookie. This is making non developers on my team have a very difficult time using previews and they are currently forced only to use previews within incognito windows to avoid this issue. 

Two types of preview related cookies I am seeing set, `io.prismic.previewSession` and `io.prismic.preview`:
![Screen Shot 2021-11-05 at 11 41 30 AM](https://user-images.githubusercontent.com/20482636/140562386-f7c21eed-a342-4cb1-8d25-187251838f12.png)
![Screen Shot 2021-11-05 at 11 41 58 AM](https://user-images.githubusercontent.com/20482636/140562387-d75fd060-2b10-44c7-bda3-68a93029f59f.png)
:

**Comment by angeloashmore** [comment]
Hi @NeversSync, apologies for the very late reply. I'm catching up on issues now.

Session cookies have somewhat unintuitive behavior. Specifically with Chrome, a session cookie is only removed when quitting the browser. If a tab or a browser window is closed without quitting Chrome, the session cookie will still be set.

Because of this, a preview session will be active until one of the following:

1. The "x" button is clicked on the Prismic Toolbar
2. The draft is published
3. The browser is quit

There are ways ensure a preview only occurs immediately after clicking the Preview button in Prismic (such as keeping state in the browser only after landing on the `/preview` page), but doing so would break Preview link sharing and auto refreshing when saving new changes. Both of these features rely on only using the presence of the preview cookie to detect an active preview session.

I'm incline to keep the current behavior and treat a preview as active until one of three actions listed above occur. Do you have any ideas of how this could work instead?

**Comment by NeversSync** [comment]
@angeloashmore Thanks for digging in to this and for the response. I can see now that I was indeed misunderstanding the expected behavior of session cookies, now that I understand that better, and the way it is enabling preview share links/live updates, I can better understand why this behavior is happening and also why it was intended. 

With that said, this behavior still feels like a broken UX, especially for non-technical users, which is the majority of people on our team actually using the preview feature. We have consistently been having the experience of a few specific 'glitches' with the preview pop up that, if resolved, may make this issue less of a pain point. If you think these issues are on our end, I'd love to know what could be causing them and figure out the fix:

1. Nearly 95% of the time when a new preview session is started, after the first few seconds, the 'fetch preview' popup is no longer clickable, making it impossible to close the preview session. This seems to be related to the browser 'crashing' as it is often accompanied by a 'page unresponsive' alert (see screenshot). Upon choosing to 'wait' the page will load. This is especially persistent and frustrating when you have previously opened a preview in a browser session and then proceed to load a non-preview page on the site, giving you the experience of being 'stuck' in a preview you didn't ask for. Any ideas what could be causing this?
2. In general, the fetch preview operation feels like it takes a very long time, the shortest I have seen, even for a simple page, is ~30 seconds, which means I'm often hearing from our team members that the preview 'broke' or 'isn't working'. I know there is a reasonable explanation to this, but I don't know what it is yet. 

<img width="501" alt="Screen Shot 2021-12-03 at 3 34 34 PM" src="https://user-images.githubusercontent.com/20482636/144686657-28c8a542-3736-4cf1-9ee3-768c8fae68ad.png">


In terms of suggestions for alternative behaviors, I don't understand enough about how the preview link sharing and the session cookie are connected to feel clear on what would work better. Would it be possible to move to a sessionStorage approach for managing the preview session? Perhaps there could be some query params added to the preview share link which, if present, would set a sessionStorage item for users to activate the preview session for them once they open the link. 
I'm not sure if that would work or not, but I'd love to collaborate to find a better alternative. Our team is essentially treating previews as broken until further notice which sucks. 


Thank you!

**Comment by mrseanbaines** [comment]
@angeloashmore any update on this? We are also experiencing all of the issues described by @NeversSync. The exceedingly long wait for the preview to load is especially annoying. From a quick look in the network tab while the preview is loading, it _looks_ like it's fetching EVERYTHING from Prismic, including documents which are completely unrelated to and not reference by the page being previewed - is that the case? We have also been experiencing a huge spike in our usage (way over our quota) and have yet to work out why, but I am wondering if this is related.

**Comment by Baztoune** [comment]
Hi guys, I found out this issue while searching for a hint. We encountered a similar performance issue in preview mode after upgrading gatsby and prismic libs to the latest versions (before that upgrade, it was slow, now it's barely usable). It looks like the preview plugin loads all our ~700 prismic documents, which can take up to several minutes on some machines. 
Is it expected or should we review our integration ? We're trying to understand this behaviour. Any hint would be much appreciated, thanks a lot !

**Comment by angeloashmore** [comment]
This issue is stale and may be fixed in a recent version. Please update to the latest version and try again. If the issue continues to be a problem, please open a new issue. Thank you!

See the documentation for guides on updating to the latest versions: [Gatsby + Prismic Migration Guides](https://github.com/prismicio/prismic-gatsby/tree/main/docs#migration-guides)
